presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -916,6 +917,12 @@ protected Type visitCurrentPath(CurrentPath node, StackableAstVisitorContext<Con
             return setExpressionType(node, VARCHAR);
         }
 
+        @Override
+        protected Type visitFormat(Format node, StackableAstVisitorContext<Context> context)
+        {
+            return setExpressionType(node, VARCHAR);
```

Comment:
Also, we should validate that the remaining arguments are supported types for format or have a cast to VARCHAR to be able to provide a more meaningful error message during analysis instead of during query optimization.

Reply:
Done